### PR TITLE
[TF] rb sparsity several replicas mask distribution

### PR DESCRIPTION
**Pull request by daniil-lyakhov**

### CODE REVIEW

**Comment by daniil-lyakhov** [comment]
To test distributed mask integrity I made test which checks equality of param `seed` for each `calc_rb_binary_mask` function call for each replica. Is it enough to confirm integrity?

**Review by alexsu52 — COMMENTED** [review]
Could you provide results for the RB sparsity algorithm?

**Comment by daniil-lyakhov** [comment]
Results of rb sparsity + int8:

|Model | Compression | FP32 acc@1 | compressed acc@1 | drop
-- | -- | -- | -- | --
InceptionV3 | sparsity 61%(rb) + int8 | 77.9 | 77.55 | 0.35
ResNet50 | sparsity 65%(rb) + int8 | 75.04 | 74.38 | 0.66
MobileNetV2 | sparsity 52%(rb) + int8 | 71.85 | 71.05 | 0.8

File: beta/nncf/tensorflow/sparsity/callbacks.py [hunk 1]
```diff
@@ -14,20 +14,24 @@
...
     def on_train_batch_begin(self, batch, logs=None):
-        self._scheduler.step()
+        self._ctrl.scheduler.step()
 
     def on_epoch_begin(self, epoch, logs=None):
-        self._scheduler.epoch_step(epoch)
+        self._ctrl.scheduler.epoch_step(epoch)
+
+    def on_test_begin(self, epoch, logs=None):
+        self._ctrl.freeze()
```
Comment (alexsu52) [inline beta/nncf/tensorflow/sparsity/callbacks.py:34]: I can assume it breaks the training pipeline because `on_test_begin` is called at the beginning of each evaluation or validation stage and the `freeze` method too.

File: beta/examples/tensorflow/classification/main.py [hunk 2]
```diff
@@ -227,7 +227,8 @@ def run(config):
         validation_dataset,
         steps=validation_steps,
         callbacks=[get_progress_bar(
-            stateful_metrics=['loss'] + [metric.name for metric in metrics])],
+            stateful_metrics=['loss'] + [metric.name for metric in metrics]),
```
Comment (alexsu52) [inline beta/examples/tensorflow/classification/main.py:230]: @daniil-lyakhov when we resume the model from the checkpoint we also load the scheduler state. It looks like the root cause is that the scheduler state is not loaded correctly. Could you take a look at this?

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: beta/examples/tensorflow/classification/main.py (hunk 2 above)
Comment (daniil-lyakhov) [reply beta/examples/tensorflow/classification/main.py:230]: According to the docstring of `load_state` base method https://github.com/openvinotoolkit/nncf/blob/develop/nncf/common/schedulers.py#L211-L217
```
    def load_state(self, state: Dict[str, object]) -> None:
        """
        Loads the compression scheduler state, but does not update the state of the
        compression method.
        :param state: Output of `get_state()` method.
        """
```
If we would update the compression parameters (target loss for instance) it would break the logic of common method.

I suggest expanding purposes of this method by compression state update for tf schedulers

File: beta/nncf/tensorflow/sparsity/callbacks.py (hunk 1 above)
Comment (daniil-lyakhov) [reply beta/nncf/tensorflow/sparsity/callbacks.py:34]: Miss it, thanks

File: beta/examples/tensorflow/classification/main.py (hunk 2 above)
Comment (daniil-lyakhov) [reply beta/examples/tensorflow/classification/main.py:230]: Besides it should solve wrong evaluation statistics issues for magnitude sparsity and filter pruning too

File: beta/examples/tensorflow/classification/configs/sparsity/inception_v3_imagenet_rb_sparsity.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply beta/examples/tensorflow/classification/configs/sparsity/inception_v3_imagenet_rb_sparsity.json]: Done

File: beta/examples/tensorflow/classification/configs/sparsity/mobilenet_v2_imagenet_rb_sparsity.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply beta/examples/tensorflow/classification/configs/sparsity/mobilenet_v2_imagenet_rb_sparsity.json]: Done

File: beta/examples/tensorflow/classification/configs/sparsity/resnet50_imagenet_rb_sparsity.json [hunk 5]
```diff
@@ -6,13 +6,13 @@
       "sample_size": [1, 224, 224, 3]
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply beta/examples/tensorflow/classification/configs/sparsity/resnet50_imagenet_rb_sparsity.json:6]: Done

File: beta/examples/tensorflow/classification/configs/sparsity_quantization/inception_v3_imagenet_rb_sparsity_int8.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply beta/examples/tensorflow/classification/configs/sparsity_quantization/inception_v3_imagenet_rb_sparsity_int8.json:1]: Done

File: beta/examples/tensorflow/classification/configs/sparsity_quantization/mobilenet_v2_imagenet_rb_sparsity_int8.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply beta/examples/tensorflow/classification/configs/sparsity_quantization/mobilenet_v2_imagenet_rb_sparsity_int8.json:1]: Done

File: beta/examples/tensorflow/classification/configs/sparsity_quantization/resnet50_imagenet_rb_sparsity_int8.json [hunk 8]
```diff
@@ -0,0 +1,36 @@
+{
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply beta/examples/tensorflow/classification/configs/sparsity_quantization/resnet50_imagenet_rb_sparsity_int8.json:1]: Done

**Review by alexsu52 — APPROVED** [review]
